sacred/experiment.py

```diff
@@ -404,7 +404,7 @@ def _gather(self, func):
         """
         for ingredient, _ in self.traverse_ingredients():
             for name, item in func(ingredient):
-                if ingredient == self:
+                if ingredient == self and len(self.path) != 0:
```

Comment:
This could be just `if ingredient == self and self.path`
as `self.path` is a string that will evaluate to `False` if it is empty.